src/unitxt/artifact.py

```diff
@@ -305,8 +305,9 @@ def save(self, path):
 
     @classmethod
     def deserialize(cls, artifact_rep):
```

Comment:
Worthwhile to document"

"Recieves a representation (either json string or dictionary)  of an Artifact and return the object."

(Maybe with while to check that it's not already an Artifact)